cxx4opencl/diff2openclc.txt

```diff
@@ -120,10 +124,47 @@ label:  // invalid: jumping forward over declaration of n
    // ...
 ----------
 
-==== Use of Clang Blocks
+==== OpenCL specific difference
+
+This section describes where {cpp} for OpenCL differs from OpenCL C
+in OpenCL specific behavior.
+
+===== Variadic macros
+
+{cpp} for OpenCL eliminates restriction on variadic macros from `OpenCL C v2.0 s6.9.e`.
+Variadic macros can be used normally as per {cpp}17 `[cpp.replace]`.
+
+===== Predefined macros
+
+Predefined version macro `__OPENCL_C_VERSION__` and macro `NULL`,
+described in `OpenCL C v2.0 s6.10`, are not supported.
```

Comment:
I'm aware that this PR is just documenting existing behaviour rather than a proposed design, so this comment can be resolved without any change, but:

Why does `C++ for OpenCL` exclude `NULL` when it is generally backwards compatible with OpenCL C 2.0?  I'm of course aware that `nullptr` is more idiomatic in modern C++, but even there [NULL is still reachable through the C headers](https://en.cppreference.com/w/cpp/types/NULL).

It just seems so easy to have something like:

```c
#define NULL nullptr
```

Reply:
Yep, I agree. One danger to that is an old definition of NULL that hasn't been the same so aliasing `NULL` to `nullptr` still doesn't provide 100% compatibility.

Of course, developers can do this trick manually if it suits their use case.

C++ for OpenCL can't achieve full backwards compatibility to OpenCL C, because it is a dialect of C++ rather than C. And in C++ there were fundamental changes that diverged from C often for good reasons I would say.

Reply:
It still seems a surprising omission to me, but it's not the point of this PR so I'll resolve this comment.